flax/nnx/module.py

```diff
@@ -493,3 +436,84 @@ def first_from(*args: tp.Optional[A], error_msg: str) -> A:
     if arg is not None:
       return arg
   raise ValueError(error_msg)
+
+def iter_modules(module: Module) -> tp.Iterator[tuple[PathParts, Module]]:
+  """Recursively iterates over all nested :class:`Module`'s of the given Module, including
+  the argument.
+
+  ``iter_modules`` creates a generator that yields the path and the Module instance, where
+  the path is a tuple of strings or integers representing the path to the Module from the
+  root Module.
+
+  Example::
+
+    >>> from flax import nnx
+    ...
+    >>> class SubModule(nnx.Module):
+    ...   def __init__(self, din, dout, rngs):
+    ...     self.linear1 = nnx.Linear(din, dout, rngs=rngs)
+    ...     self.linear2 = nnx.Linear(din, dout, rngs=rngs)
+    ...
+    >>> class Block(nnx.Module):
+    ...   def __init__(self, din, dout, *, rngs: nnx.Rngs):
+    ...     self.linear = nnx.Linear(din, dout, rngs=rngs)
+    ...     self.submodule = SubModule(din, dout, rngs=rngs)
+    ...     self.dropout = nnx.Dropout(0.5)
+    ...     self.batch_norm = nnx.BatchNorm(10, rngs=rngs)
+    ...
+    >>> model = Block(2, 5, rngs=nnx.Rngs(0))
+    >>> for path, module in iter_modules(model):
```

Comment:
```suggestion
    >>> for path, module in nnx.iter_modules(model):
```

Put `nnx.` everywhere you references `iter_modules`